Find a functionally equivalent code:android.os.SystemClock.setCurrentTimeMillis:COMMENT
<android.os.SystemClock: boolean setCurrentTimeMillis(long)>
Method Modifier: public      static      native      
Method Modifier: Yes         Yes         Yes         
Method Updated:  Yes         Yes         No          
Comment:/**
 * Sets the current wall time, in milliseconds.  Requires the calling
 * process to have appropriate permissions.
 *
 * @return if the clock was successfully set to the specified time.
 */

@@ -1 +1,15 @@
-
+{
+    IBinder b = ServiceManager.getService(Context.ALARM_SERVICE);
+    IAlarmManager mgr = IAlarmManager.Stub.asInterface(b);
+    if (mgr == null) {
+        return false;
+    }
+    try {
+        return mgr.setTime(millis);
+    } catch (RemoteException e) {
+        Slog.e(TAG, "Unable to set RTC", e);
+    } catch (SecurityException e) {
+        Slog.e(TAG, "Unable to set RTC", e);
+    }
+    return false;
+}

